spaghetto/bin/src/Flags.purs

```diff
@@ -4,13 +4,87 @@ import Spago.Prelude
 
 import ArgParse.Basic (ArgParser)
 import ArgParse.Basic as ArgParser
+import Data.Set.NonEmpty (NonEmptySet)
+import Data.Set.NonEmpty as NonEmptySet
+import Node.Path as Path
+import Spago.Core.Config (ShowSourceCode(..))
+import Spago.Core.Config as Core
+import Spago.Paths as Paths
 
 selectedPackage ∷ ArgParser (Maybe String)
 selectedPackage =
   ArgParser.argument [ "--package", "-p" ]
     "Select the local project to build"
     # ArgParser.optional
 
+psaStrict ∷ ArgParser (Maybe Boolean)
+psaStrict =
+  ArgParser.flag [ "--strict" ]
+    "Promotes project sources' warnings to errors"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaCensorWarnings ∷ ArgParser (Maybe Boolean)
+psaCensorWarnings =
+  ArgParser.flag [ "--psa-censor-warnings" ]
+    "Censor all warnings"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaCensorLib ∷ ArgParser (Maybe Boolean)
+psaCensorLib =
+  ArgParser.flag [ "--psa-censor-lib" ]
+    "Censor warnings from library sources"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaCensorSrc ∷ ArgParser (Maybe Boolean)
+psaCensorSrc =
+  ArgParser.flag [ "--psa-censor-src" ]
+    "Censor warnings from project sources"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaShowSource ∷ ArgParser (Maybe ShowSourceCode)
+psaShowSource =
+  NoSourceCode
+    <$ ArgParser.flag [ "--psa-no-source" ]
+      "Disable original source code printing"
+    # ArgParser.optional
```

Comment:
I used an ADT here rather than a Boolean as it seemed a bit easier to read because the existence of this flag disables the 'show source code' feature.

Reply:
That's alright - the codebases uses booleans all over to avoid conversions, but this works too